application_form/services/application.py

```diff
@@ -344,5 +347,13 @@ def _find_winning_candidates(applications: QuerySet) -> QuerySet:
     """Return all applications that have the smallest right of residence number."""
     if not applications.exists():
         return applications.none()
-    min_right_of_residence = applications.first().right_of_residence
-    return applications.filter(right_of_residence=min_right_of_residence)
+
+    first_app = applications.first()
```

Comment:
The code used to rely on the order of the applications given in the queryset parameter or possibly is implicitly using the default ordering of the application model. If it's using the default ordering, did you check that the old-batch boolean is accounted for there?

Reply:
It seems that the applications provided via the argument are already ordered by the `queue_position` field. So there's no problem with this.